lua/telescope/builtin/files.lua

```diff
@@ -261,15 +277,155 @@ files.file_browser = function(opts)
           Path:new(fpath):touch({ parents = true })
           vim.cmd(string.format(':e %s', fpath))
         else
+          -- TODO(conni2461): I think when doing realpath we don't have to worry about :sub(1, -2) anymore
           Path:new(fpath:sub(1, -2)):mkdir({ parents = true })
-          local new_cwd = vim.fn.expand(fpath)
+          local new_cwd = vim.loop.fs_realpath(fpath)
           current_picker.cwd = new_cwd
-          current_picker:refresh(gen_new_finder(new_cwd), { reset_prompt = true })
+          current_picker:refresh(gen_new_finder({
+            path = new_cwd,
+          }), { reset_prompt = true })
+        end
+      end
+
+      local rename_file = function()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+        local old_name = Path:new(action_state.get_selected_entry()[1])
+
+        if old_name.filename == '../' then
+          print('Please select a file!')
+          return
+        end
+
+        local new_name = vim.fn.input("Insert a new name: ", old_name:make_relative())
+
+        old_name:rename({ new_name = new_name })
+        current_picker:refresh(gen_new_finder({
+          path = current_picker.cwd,
+        }), { reset_prompt = true })
+      end
+
+      local get_marked_files = function ()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+        local multi_selected = current_picker:get_multi_selection()
+        local entries
+
+        if vim.tbl_isempty(multi_selected) then
+          entries = { action_state.get_selected_entry() }
+        else
+          entries = multi_selected
+        end
+
+        local selected = vim.tbl_map(function(entry)
+          return Path:new(entry[1])
+        end, entries)
+
+        return selected
+      end
+
+      local move_file = function()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+
+        for _, file in ipairs(get_marked_files()) do
+          local filename = file.filename:sub(#file:parents() + 2)
+
+          local success = file:rename({
+            new_name = Path:new({ current_picker.cwd, filename }).filename
+          })
+
+          if not success then
+            print("The file is already exists!")
+            return
+          end
         end
+
+        print("The file has been moved!")
+        current_picker:refresh(gen_new_finder({
+          path = current_picker.cwd,
+        }), { reset_prompt = true })
+      end
+
+      local copy_file = function()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+
+        for _, file in ipairs(get_marked_files()) do
+          local filename = file.filename:sub(#file:parents() + 2)
+
+          local success = file:copy({
+            destination = Path:new({
+              current_picker.cwd, filename
+            }).filename
+          })
+
+          if not success then
+            print("The file is already exists!")
```

Comment:
This can cause a *false alarm*, eg: it's copied succesfully but it prints this warning
removing this fixed it so I'm gonna remove it. Not sure if removing it is a good idea or not.